### PR TITLE
don't allocate vgpr in globalWriteBatch

### PR DESCRIPTION
**Pull request by TonyYHsieh**
globalWriteBatch will allocate most of vgpr in order to do batch write.
So allocate vgpr in globalWriteBatch is easily to met vpgr overflow.
We should allocate all needed vgpr outside of globalWriteBatch.

### CODE REVIEW

**Comment by rosenrodt** [comment]
Can we have a check to disallow checkouts of vgprs during store, in case someone in the future doesn't know what kind of trap he'll walk into?

**Comment by TonyYHsieh** [comment]
> Can we have a check to disallow checkouts of vgprs during store, in case someone in the future doesn't know what kind of trap he'll walk i

**Comment by TonyYHsieh** [comment]
> Can we have a check to disallow checkouts of vgprs during store, in case someone in the future doesn't know what kind of trap he'll walk into?

not in this PR.

File: Tensile/KernelWriterAssembly.py [hunk 1]
```diff
@@ -10623,6 +10626,8 @@ def globalWriteElements(self, kernel, vectorWidths, elements,
     # End label
     kStr += "label_%s:%s"%(endLabel, self.endLine)
     self.vgprPool.checkIn(tmpVgpr)
+    if bf16CVTVgpr != None:
```
Comment (cgmillette) [inline Tensile/KernelWriterAssembly.py:10629]: Should probably use 
if bf16CVTVgpr is not None

**Review by cgmillette — APPROVED** [review]
I suggest using "is not None" vs "!= None".

Otherwise, LGTM, as long as CI is happy.

File: Tensile/KernelWriterAssembly.py (hunk 1 above)
Comment (TonyYHsieh) [reply Tensile/KernelWriterAssembly.py]: update

**Comment by zaliu** [comment]
@TonyYHsieh please resolve the source conflict.  Thanks.

**Pull request merged**